scripts/deprecate-old-versions.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+# Some inspiration drawn from Apollo Server's prior art:
+# https://github.com/apollographql/apollo-server/blob/5d87be74ca68376b6698fed28adeaccdfcb639ef/scripts/deprecate-old-versions.sh
+
+# As far as we can tell, `npm deprecate` only deprecates currently existing
+# package versions. Whenever we publish a backport to v0.x, the latest
+# version will end up not deprecated! So after publishing backport versions, we
+# run this script (with apollo-bot credentials). As far as we can tell you can't
+# use npm tokens for this so it requires you to enter a ton of OTPs. Fun!
+# https://stackoverflow.com/questions/74466186/keeping-npm-deprecated-packages-deprecated
+
+set -e
+
+# List of packages to deprecate
+packages=("@apollo/gateway" "@apollo/federation" "@apollo/query-planner" "@apollo/subgraph")
+
+for package_name in "${packages[@]}"
+do
+  # Get all the versions of the package, trim double quotes
+  versions=$(npm view $package_name versions --json | tr -d '[],"')
+  for version in $versions; do
+    # Filter for versions starting with 0.x only
+    if [[ $version == 0.* ]]; then
+      # Print the version
+      echo "Deprecating version: $version of package $package_name"
+      # Deprecate the version
+      npm deprecate "$package_name@$version" "The `$package_name` package is part of Federation v1, which is now deprecated (end-of-life September 22nd 2023). Please upgrade your package to its latest counterpart. See announcement blog for details (https://www.apollographql.com/blog/announcement/backend/announcing-the-end-of-life-schedule-for-apollo-gateway-v0-x/)."
```

Comment:
My suggestion to have the following deprecation message: 

"All 0.x versions of @apollo/gateway are now deprecated (end-of-life September 22, 2023). Please upgrade to v2.x or migrate to the Apollo Router. See[ announcement blog](https://www.apollographql.com/blog/announcement/backend/announcing-the-end-of-life-schedule-for-apollo-gateway-v0-x/) or[ documentation](https://www.apollographql.com/docs/federation/federation-2/backward-compatibility/#is-official-support-ending-for-apollogateway-v0x) for more details."

Reply:
That sounds good to me for the gateway's message. What do you suggest for the non-gateway packages that are v0?
>"@apollo/federation" "@apollo/query-planner" "@apollo/subgraph"

It might be a bit nuanced / might make sense to have a different message for each:
* If people are still using @apollo/federation they should probably be using @apollo/subgraph instead. It's possible they're using it for composition as well but that's an unexpected use case.
* @apollo/query-planner is an internal gateway package that people shouldn't be using (which seems to be the case, roughly https://www.npmjs.com/package/@apollo/query-planner?activeTab=dependents)
* @apollo/subgraph is...for subgraphs so it doesn't really make sense to say "use the router" in its deprecation message